src/com/google/enterprise/adaptor/DocumentContentTransform.java

```diff
@@ -0,0 +1,79 @@
+package com.google.enterprise.adaptor;
+
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Collections;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * The document content transform can modify the content of a document.
+ *
+ * @author Dominik Weidenfeld (dominik.weidenfeld@twt.de)
+ */
+public class DocumentContentTransform extends OutputStream {
+
+  protected final Map<String, String> config;
+  protected Metadata metadata;
+  protected String contentType;
+  private OutputStream originalStream;
+
+  /**
+   * Constructs a document content transform.
+   *
+   * @param config the configuration for this instance
+   */
+  public DocumentContentTransform(final Map<String, String> config) {
+    this.config = Collections.unmodifiableMap(
+        new TreeMap<String, String>(config));
+  }
+
+  public final void setMetadata(final Metadata metadata) {
+    this.metadata = metadata;
+  }
+
+  public final void setContentType(final String contentType) {
+    this.contentType = contentType;
+  }
+
+  public final void setOriginalStream(final OutputStream originalStream) {
```

Comment:
It was made because of our previous change (newInstance in ctor). That time there was no other way to pass the information to this class.
Now I added everything to the ctor again, because the newInstance is not called in the ctor anymore.